Update documentation for date and h3 features. (#68)
ludwig-ai/ludwig-docs: docs/configuration/features/date_features.md

  datetime_format: "%d %b %Y"
```

Ludwig will try to infer the date format automatically, but a specific format can be provided. The date string spec is
the same as the one described in python's [datetime](https://docs.python.org/3/library/datetime.html#strftime-strptime-behavior).

Here are some preprocessing parameters.

- `missing_value_strategy` (default `fill_with_const`): what strategy to follow when there's a missing value. The value should be one of `fill_with_const` (replaces the missing value with a specific value specified with the `fill_value` parameter). The special value `backfill` replaces the missing values with the next valid value.

> - `fill_value` (default `""`): the value to replace missing values with when `missing_value_strategy` is `fill_value`. This can be a datetime string. If empty, the current datetime will be used.

Reviewer: ```suggestion
- `fill_value` (default `""`): the value to replace missing values with when `missing_value_strategy` is `fill_with_const`. This can be a datetime string. If empty, the current datetime will be used.
```

use_bias: true
weights_initializer: glorot_uniform
bias_initializer: zeros
norm: null
norm_params: null
activation: relu
dropout: 0
```

This encoder passes the year through a fully connected layer of one neuron and embeds all other elements for the date,
concatenates them and passes the concatenated representation through fully connected layers.

It takes the following optional parameters:

> - `embedding_size` (default `10`): it is the maximum embedding size adopted..

Reviewer: ```suggestion
- `embedding_size` (default `10`): it is the maximum embedding size adopted.
```

- `embeddings_on_cpu` (default `false`): by default embeddings matrices are stored on GPU memory if a GPU is used, as it allows for faster access, but in some cases the embedding matrix may be really big and this parameter forces the placement of the embedding matrix in regular memory and the CPU is used to resolve them, slightly slowing down the process as a result of data transfer between CPU and GPU memory.

> - `dropout` (default `false`): determines if there should be a dropout layer before returning the encoder output.

Reviewer: ```suggestion
- `dropout` (default `0`): dropout rate.
```

> - `fc_layers` (default `null`): it is a list of dictionaries containing the parameters of all the fully connected layers. The length of the list determines the number of stacked fully connected layers and the content of each dictionary determines the parameters for a specific layer. The available parameters for each layer are: `fc_size`, `norm`, `activation` and `regularize`. If any of those values is missing from the dictionary, the default one specified as a parameter of the encoder will be used instead. If both `fc_layers` and `num_fc_layers` are `null`, a default list will be assigned to `fc_layers` with the value `[{fc_size: 512}, {fc_size: 256}]` (only applies if `reduce_output` is not `null`).

Reviewer: ```suggestion
a list of dictionaries containing the parameters of all the fully connected
layers. The length of the list determines the number of stacked fully connected layers and the content of each
dictionary determines the parameters for a specific layer. The available parameters for each layer are: `activation`,
`dropout`, `norm`, `norm_params`, `output_size`, `use_bias`, `bias_initializer` and `weights_initializer`. If any of
those values is missing from the dictionary, the default one specified as a parameter of the encoder will be used
instead.```

- `num_fc_layers` (default `0`): This is the number of stacked fully connected layers.

> - `fc_size` (default `10`): if a `fc_size` is not already specified in `fc_layers` this is the default `fc_size` that will be used for each layer. It indicates the size of the output of a fully connected layer.

Reviewer: ```suggestion
- `output_size` (default `10`): if a `output_size` is not already specified in `fc_layers` this is the default `output_size` that will be used for each layer. It indicates the size of the output of a fully connected layer.
```

use_bias: true
weights_initializer: glorot_uniform
bias_initializer: zeros
norm: null
norm_params: null
activation: relu
dropout: 0
```

This encoder passes the year through a fully connected layer of one neuron and represents all other elements for the date by taking the cosine of their value with a different period (12 for months, 31 for days, etc.), concatenates them and passes the concatenated representation through fully connected layers.

It takes the following parameters:

> - `fc_layers` (default `null`): it is a list of dictionaries containing the parameters of all the fully connected layers. The length of the list determines the number of stacked fully connected layers and the content of each dictionary determines the parameters for a specific layer. The available parameters for each layer are: `fc_size`, `norm`, `activation` and `regularize`. If any of those values is missing from the dictionary, the default one specified as a parameter of the encoder will be used instead. If both `fc_layers` and `num_fc_layers` are `null`, a default list will be assigned to `fc_layers` with the value `[{fc_size: 512}, {fc_size: 256}]` (only applies if `reduce_output` is not `null`).

Reviewer: ```suggestion
- `fc_layers` (default `null`): a list of dictionaries containing the parameters of all the fully connected
layers. The length of the list determines the number of stacked fully connected layers and the content of each
dictionary determines the parameters for a specific layer. The available parameters for each layer are: `activation`,
`dropout`, `norm`, `norm_params`, `output_size`, `use_bias`, `bias_initializer` and `weights_initializer`. If any of
those values is missing from the dictionary, the default one specified as a parameter of the encoder will be used
instead.
```

- `num_fc_layers` (default `0`): This is the number of stacked fully connected layers.

> - `fc_size` (default `10`): if a `fc_size` is not already specified in `fc_layers` this is the default `fc_size` that will be used for each layer. It indicates the size of the output of a fully connected layer.

Reviewer: ```suggestion
- `output_size` (default `10`): if a `output_size` is not already specified in `fc_layers` this is the default `output_size` that will be used for each layer. It indicates the size of the output of a fully connected layer.
```
